activejob/lib/active_job/logging.rb

```diff
@@ -87,12 +88,25 @@ def queue_name(event)
         def args_info(job)
           if job.arguments.any?
             ' with arguments: ' +
-              job.arguments.map { |arg| arg.try(:to_global_id).try(:to_s) || arg.inspect }.join(', ')
+              job.arguments.map { |arg| format(arg).inspect }.join(', ')
           else
             ''
           end
         end
 
+        def format(arg)
+          case arg
+          when Hash
+            arg.transform_values { |value| format(value) }
+          when Array
+            arg.map { |value| format(value) }
+          when GlobalID::Identification
+            begin arg.to_global_id rescue arg end
```

Comment:
I sure hope `to_global_id` raises for an unpersisted model, that's why we added that error: https://github.com/rails/globalid/pull/68 :wink:

Also, you don't need `begin` here. This will do:

```ruby
arg.to_global_id rescue arg
```